### PR TITLE
Making upper bound inclusive

### PR DESCRIPTION
**Pull request by bouthilx**
Why:

The priors typically have exclusive upper bound but due to numerical
errors algorithms are likely to suggest points on the upper bound.

### CODE REVIEW

**Comment by bouthilx** [comment]
[Fix out-of-space issue](https://trello.com/c/WYLirJEi/473-fix-out-of-space-issue)

File: codecov.yml [hunk 1]
```diff
@@ -40,7 +40,8 @@ coverage:
       default:
         branches: null
 
-  ignore: null
+  ignore:
+    - tests/*
```
Comment (Thomsch) [inline codecov.yml:44]: This configuration change is outside the scope of this bugfix. Can I suggest, we add those changes in a separate PR?

File: src/orion/algo/space.py [hunk 2]
```diff
@@ -312,44 +310,32 @@ def _is_numeric(item):
...
-    high : float
-       Constrain with an upper bound (exclusive), default ``numpy.inf``.
+       Lower bound (inclusive), optional; default ``-numpy.inf``.
+    high : float:
+       Upper bound (inclusive), optional; default ``numpy.inf``.
+    precision : int
+        Precision, optional; default ``4``.
+    shape : tuple
+       Defines how many dimensions are packed in this `Dimension`.
+       Describes the shape of the corresponding tensor.
```
Comment (Thomsch) [inline src/orion/algo/space.py:335]: Since the expected behavior is that the higher bound is exclusive, maybe we could add a quick rationale for why it's not the case.

File: src/orion/algo/space.py [hunk 3]
```diff
@@ -475,7 +461,7 @@ def interval(self, alpha=1.0):
             int_high = int(numpy.floor(high))
         except OverflowError:  # infinity cannot be converted to Python int type
             int_high = numpy.inf
-        if int_high < high:  # Exclusive upper bound
+        if int_high < high:  # Inclusive upper bound
```
Comment (Thomsch) [inline src/orion/algo/space.py:464]: Why is it not `<=` ?

File: src/orion/algo/space.py [hunk 4]
```diff
@@ -484,15 +470,28 @@ def __contains__(self, point):
 
 
 class Integer(Real, _Discrete):
-    """Subclass of `Dimension` for representing integer parameters.
+    """Search space dimension that can take on any real value.
```
Comment (Thomsch) [inline src/orion/algo/space.py:473]: > any real value.
What is the difference compared to the base class `Real`?

File: codecov.yml (hunk 1 above)
Comment (bouthilx) [reply codecov.yml:44]: Fair enough. I created #377 and will now remove it from this PR.

File: src/orion/algo/space.py (hunk 3 above)
Comment (bouthilx) [reply src/orion/algo/space.py:464]: Because we would overflow to next integer when `int_high == high`. That should be removed actually and use `ceil` instead of `floor` for `int_high`.

File: src/orion/algo/space.py (hunk 4 above)
Comment (bouthilx) [reply src/orion/algo/space.py:473]: Oopsie. Good catch, that was wrong copy paste from Real class.

File: src/orion/algo/space.py (hunk 2 above)
Comment (bouthilx) [reply src/orion/algo/space.py:335]: Do you mean in the docstring for the parameter `high` or as a note?

File: codecov.yml (hunk 1 above)
Comment (Thomsch) [reply codecov.yml]: Thanks!

File: src/orion/algo/space.py (hunk 2 above)
Comment (Thomsch) [reply src/orion/algo/space.py:335]: I was thinking to include it in the class's docstring.

File: src/orion/algo/space.py (hunk 3 above)
Comment (Thomsch) [reply src/orion/algo/space.py]: Thanks for the explanations!
